flair/datasets/biomedical.py

```diff
@@ -188,7 +187,9 @@ def filter_nested_entities(dataset: InternalBioNerDataset) -> None:
     num_entities_after = sum([len(x) for x in dataset.entities_per_document.values()])
     if num_entities_before != num_entities_after:
         removed = num_entities_before - num_entities_after
-        warn(f"Corpus modified by filtering nested entities. Removed {removed} entities.")
+        warn(
+            f"Corpus modified by filtering nested entities. Removed {removed} entities. Keep {num_entities_after} entities."
+        )
```

Comment:
This warning prints oddly for me on first download. It prints 

```console
UserWarning: Corpus modified by filtering nested entities. Removed 63 entities. Keep 3548 entities.
  f"Corpus modified by filtering nested entities. Removed {removed} entities. Keep {num_entities_after} entities."
```

Not a big deal, but a bit strange that the f-string is printed as well. Can you check if this is happening on your end as well?

Reply:
Yeah, happening on my end as well. Seems to depend the format of the warning messages, see https://stackoverflow.com/questions/2187269/print-only-the-message-on-warnings

I would just use logger.warning(...) instead of warnings.warn(...) as the logging tutorial recommends. The warning is an informative message and not a cause to modify the application, see https://stackoverflow.com/questions/9595009/warnings-warn-vs-logging-warning/14762106#14762106

How is this handled elsewhere in Flair?

Reply:
Yes, we use the logger for warnings normally (log.warning(..)). See for instance https://github.com/flairNLP/flair/blob/117e14e5d821a1a4a655ccbbe495fc380ecebb7a/flair/datasets/sequence_labeling.py#L827

Reply:
Alright, I've changed the corresponding places in the code to logger.warning(...) instead.